runtime/compiler/control/CompilationThread.cpp

```diff
@@ -4708,7 +4708,7 @@ bool TR::CompilationInfo::isQueuedForCompilation(J9Method * method, void *oldSta
    return linkageInfo->isBeingCompiled();
    }
 
-#if defined(OSX)
+#if defined(OSX) || (defined(LINUX) && defined(TR_HOST_64BIT))
```

Comment:
@0xdaryl 
I agree that doing a check for `NASM_ASSEMBLER` is a lot easier to understand rather than these platform checks in so many places. There are 2 ways we could define `NASM_ASSEMBLER`:
1. Change Makefile (and CMakefile) variables to add `NASM_ASSEMBLER` as one of the defines for ALL C++ source files.
2. Modify [`asmprotos.h`](https://github.com/eclipse/openj9/blob/master/runtime/compiler/runtime/asmprotos.h) to define `NASM_ASSEMBLER` for the enabled platforms.

I will go with option 2 as I think that's obviously a much cleaner approach.

Reply:
This would also mean that once the switch is thrown we could go back to the old assembler procedure until the old code paths are deprecated? That would be an ideal state so we can let the changes sit for a few weeks as the default before purging the old code and having the ability to do a build to compare against for any functional issues we might see...

Reply:
@andrewcraik Unfortunately falling back using the old code is not going to be simply removing this macro. `jilconsts.inc` is set to generate in NASM format on x86-64 Linux due to the changes [here](https://github.com/eclipse/openj9/pull/3293/files#diff-d11f76249f168c90166a05aeb3604204R78) and [here](https://github.com/eclipse/openj9/pull/3293/files#diff-d11f76249f168c90166a05aeb3604204L95). Furthermore, to use the old files in x86-64 Linux, you'd also need to remove [this](https://github.com/eclipse/openj9/pull/3293/files#diff-cd91d09ca63960b340fd6701c8b636d4R27) line from one of the Makefiles.